bin/nbb

```diff
@@ -344,6 +339,24 @@ find_centos_7_ami_ids() {
   done
 }
 
+shutdown_instances() {
+  aws ec2 describe-instances --filters "Name=instance-state-name,Values=running" | jq -c -r '.Reservations[] .Instances[] .InstanceId' > instance_list
+  echo "You are about to terminate the following instances:"
+  cat instance_list
+  read -r -p "Are you sure? [y/N] " response
+  if [[ $response =~ ^([yY][eE][sS]|[yY])$ ]]
+  then
+    IFS=$'\n'  # make newlines the separator
+    for i in `cat ./instance_list`; do
+      echo "Shutting down $i"
+      aws ec2 terminate-instances --instance-ids $i
```

Comment:
I think you might not have been present for the conversations we've had about this over the last couple weeks (just catching you up, not an accusatory statement!). Basically it boils down to this: `make destroy` calls Terraform's destroy command which is **supposed** to cascade down, _deleting all the things inside the VPCs, subnets, etc. automatically_.

The problem is that while `terraform` seems to be working fine, **Amazon** isn't behaving according to its own API. That cascade effect winds up failing roughly half the time because Amazon's API throws Dependency Violation errors, meaning that it refuses to delete an object because something still exists that depends on it. I've personally seen this at least a dozen times just myself; I'm sure the rest of the team has too.

So basically it boils down to the fact that we need to be able to see, _outside of `terraform`_, what's running (those "dependent" objects) so we can be sure to terminate them properly. **THEN** `terraform destroy` can actually work because the Amazon API won't barf on us.

💩